datamimic_ce/tasks/generate_task.py

```diff
@@ -834,88 +405,53 @@ def execute(self, context: SetupContext | GenIterContext) -> dict[str, list] | N
 
-        # Generate and export if gen_stmt is outermost (which has context as SetupContext)
-        exporter_util = context.root.class_factory_util.get_exporter_util()
-        if isinstance(context, SetupContext):
-            consumer_with_operations, consumer_without_operations = exporter_util.create_exporter_list(
-                setup_context=context,
-                stmt=self.statement,
-                targets=self.statement.targets,
-                page_info=None,
-            )
-            # Check for conditions to use multiprocessing
-            has_mongodb_delete = any(
-                [
-                    operation == "delete" and isinstance(consumer, MongoDBExporter)
-                    for consumer, operation in consumer_with_operations
-                ]
-            )
-            match self.statement.multiprocessing:
-                case None:
-                    use_mp = (not has_mongodb_delete) and context.use_mp
-                case _:
-                    use_mp = bool(self.statement.multiprocessing)
-
-            # Generate in multiprocessing
-            if use_mp:
-                # IMPORTANT: always use deep copied setup_ctx for mp to avoid modify original setup_ctx accidentally
-                copied_ctx = copy.deepcopy(context)
-                # Process data by page
-                logger.info(f"Processing by page with size {page_size} for '{self.statement.name}'")
-                self._mp_page_process(
-                    copied_ctx,
-                    page_size,
-                    _geniter_single_process_generate_and_consume_by_page,
-                )
-            # Generate and consume in single process
-            else:
-                # Process data by page in single process
-                index_chunk = self._get_chunk_indices(page_size, count)
-                logger.info(f"Processing {len(index_chunk)} pages for {count:,} products of '{self.statement.name}'")
-                for page_index, page_tuple in enumerate(index_chunk):
-                    start, end = page_tuple
-                    logger.info(
-                        f"Processing {end - start:,} product '{self.statement.name}' on page "
-                        f"{page_index + 1}/{len(index_chunk)} in a single process"
-                    )
-                    # Generate product
-                    result = self._sp_generate(context, start, end)
-                    # Consume by page
-                    _consume_by_page(
-                        self.statement,
-                        context,
-                        result,
-                        page_index,
-                        page_size,
-                        None,
-                        None,
-                        page_index == len(index_chunk) - 1,
-                    )
+            # Determine number of Ray workers
+            num_workers = int(self.statement.num_process or context.root.num_process or 1)
```

Comment:
yeah ... this is how it should be ... default is num_process = 1 ... we need to make sure the ray framework is doing what we expect ... last time even when i specified 1 process ... many ray processes were created without any load ... haven't investigated more if this behavior is normal and how to address